### PR TITLE
fix Scharr nomination

### PR DESCRIPTION
**Pull request by PetWorm**
### This merge aim to fix a historical nominating mistake

Resolves issue [#17175](https://github.com/opencv/opencv/issues/17175)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by PetWorm** [comment]
Notice that in 3.4, 'tegra::calcSharrDeriv' is used if HAVE_TEGRA_OPTIMIZATION. This function is in Opencv4Tegra and I think it is a subsequent mistake following original OpenCV. I'm new to this community so I don't know if there is any way to fix that.

File: modules/video/src/lkpyramid.cpp [hunk 1]
```diff
@@ -803,7 +803,7 @@ int cv::buildOpticalFlowPyramid(InputArray _img, OutputArrayOfArrays pyramid, Si
                 deriv.create(sz.height + winSize.height*2, sz.width + winSize.width*2, derivType);
 
             Mat derivI = deriv(Rect(winSize.width, winSize.height, sz.width, sz.height));
-            calcSharrDeriv(thisLevel, derivI);
+            calccScharrDeriv(thisLevel, derivI);
```
Comment (dkurt) [inline modules/video/src/lkpyramid.cpp:806]: There is a type here: `calcc`

Comment (PetWorm) [reply modules/video/src/lkpyramid.cpp]: sorry, my bad, fixed

File: modules/video/src/lkpyramid.cpp [hunk 2]
```diff
@@ -1384,7 +1384,7 @@ void SparsePyrLKOpticalFlowImpl::calc( InputArray _prevImg, InputArray _nextImg,
             Mat _derivI( imgSize.height + winSize.height*2,
                 imgSize.width + winSize.width*2, derivIBuf.type(), derivIBuf.ptr() );
             derivI = _derivI(Rect(winSize.width, winSize.height, imgSize.width, imgSize.height));
-            calcSharrDeriv(prevPyr[level * lvlStep1], derivI);
+            calcScharrDeriv(prevPyr[level * lvlStep1], derivI);
             copyMakeBorder(derivI, _derivI, winSize.height, winSize.height, winSize.width, winSize.width, BORDER_CONSTANT|BORDER_ISOLATED);
         }
         else
```
Comment (dkurt) [inline modules/video/src/lkpyramid.cpp:1390]: grep shows there there should be 6 entries:
```
$ grep -r "Sharr" ~/opencv
./modules/video/src/lkpyramid.cpp:static void calcSharrDeriv(const cv::Mat& src, cv::Mat& dst)
./modules/video/src/lkpyramid.cpp:    parallel_for_(Range(0, rows), cv::detail::SharrDerivInvoker(src, dst), cv::getNumThreads());
./modules/video/src/lkpyramid.cpp:void cv::detail::SharrDerivInvoker::operator()(const Range& range) const
./modules/video/src/lkpyramid.cpp:    if (tegra::useTegra() && tegra::calcSharrDeriv(src, dst))
./modules/video/src/lkpyramid.cpp:            calcSharrDeriv(thisLevel, derivI);
./modules/video/src/lkpyramid.cpp:            calcSharrDeriv(prevPyr[level * lvlStep1], derivI);
```

Comment (dkurt) [reply modules/video/src/lkpyramid.cpp:1390]: Oh, I see, I've already mentioned `tegra::calcSharrDeriv`

**Pull request merged**